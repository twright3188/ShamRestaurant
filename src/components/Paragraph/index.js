import React from 'react';
import PropTypes from 'prop-types';
import { Text, StyleSheet } from 'react-native';
import Colors from '../../theme';

const styles = StyleSheet.create({
  paragraphStyles: {
    fontFamily: Colors.FontBaseRegular,
    fontWeight: '400',
    flexWrap: 'wrap',
  }
});

function Paragraph(props) {
  const { children, fontSize, color, style } = props;

  return (
    <Text numberOfLines={2} style={[styles.paragraphStyles, { fontSize , color , lineHeight: fontSize * 1.5 }, style]}>
      {children}
    </Text>
  );
}

Paragraph.defaultProps = {
  fontSize: 16,
  children: '',
  color: Colors.text,
  style: {}
};

Paragraph.propTypes = {
  fontSize: PropTypes.number,
  children: PropTypes.node,
  color: PropTypes.string,
  style: PropTypes.shape({ root: PropTypes.string })
};

export default Paragraph;