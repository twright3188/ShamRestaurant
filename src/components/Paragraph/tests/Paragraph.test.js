import React from 'react';
import renderer from 'react-test-renderer';
import Paragraph from '../index';

it('paragraph render', () => {
    const tree = renderer.create(<Paragraph>test</Paragraph>).toJSON();
    expect(tree).toMatchSnapshot();
});


