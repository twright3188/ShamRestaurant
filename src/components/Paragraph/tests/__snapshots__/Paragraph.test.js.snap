// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`paragraph render 1`] = `
<Text
  style={
    Array [
      Object {
        "flexWrap": "wrap",
        "fontFamily": "Helvetica",
        "fontWeight": "400",
      },
      Object {
        "color": "#222624",
        "fontSize": 16,
        "lineHeight": 24,
      },
      Object {},
    ]
  }
>
  test
</Text>
`;
